telluric/vrt.py

```diff
@@ -5,6 +5,89 @@
 from rasterio.dtypes import _gdal_typename
 from rasterio.enums import MaskFlags
 from rasterio.path import parse_path, vsi_path
+from rasterio.crs import CRS
+from rasterio.io import MemoryFile
+from rasterio.windows import from_bounds
+import os
+from xml.dom import minidom
+
+
+def prettify(elem):
+    """Return a pretty-printed XML string for the Element.
+    """
+    rough_string = ET.tostring(elem, 'utf-8')
+    reparsed = minidom.parseString(rough_string)
+    return reparsed.toprettyxml(indent="\t")
+
+
+def find_and_convert_to_type(_type, node, path):
+    value = node.find(path)
+    if value:
+        value = _type(value)
+    return value
+
+
+def wms_vrt(wms_file, bounds=None, resolution=None):
+    from telluric import rasterization, constants
+    wms_tree = ET.parse(wms_file)
+    left = find_and_convert_to_type(float, wms_tree, ".//DataWindow/UpperLeftX")
+    up = find_and_convert_to_type(float, wms_tree, ".//DataWindow/UpperLeftY")
+    right = find_and_convert_to_type(float, wms_tree, ".//DataWindow/LowerRightX")
+    bottom = find_and_convert_to_type(float, wms_tree, ".//DataWindow/LowerRightY")
+    src_bounds = (left, bottom, right, up)
+    bounds = bounds or src_bounds
+    src_resolution = constants.MERCATOR_RESOLUTION_MAPPING[20]
+    resolution = resolution or constants.MERCATOR_RESOLUTION_MAPPING[20]
+    dst_height, dst_width, transform = rasterization.raster_data(bounds=bounds, dest_resolution=resolution)
+    orig_height, orig_width, orig_transform = rasterization.raster_data(
+        bounds=src_bounds, dest_resolution=src_resolution)
+    src_window = from_bounds(*bounds, orig_transform)
+    vrtdataset = ET.Element('VRTDataset')
+    vrtdataset.attrib['rasterXSize'] = str(dst_height)
+    vrtdataset.attrib['rasterYSize'] = str(dst_width)
+    srs = ET.SubElement(vrtdataset, 'SRS')
+    projection = find_and_convert_to_type(str, wms_tree, ".//projection")
+    blockx = find_and_convert_to_type(str, wms_tree, "//BlockSizeX")
+    blocky = find_and_convert_to_type(str, wms_tree, "//BlockSizeY")
+    projection = CRS(init=projection)
+    srs.text = projection.wkt
+    geotransform = ET.SubElement(vrtdataset, 'GeoTransform')
+    geotransform.text = ','.join([str(v) for v in transform.to_gdal()])
+    image_metadata = ET.SubElement(vrtdataset, 'Metadata')
+    image_metadata.attrib["domain"] = "IMAGE_STRUCTURE"
+    image_mdi = ET.SubElement(image_metadata, "MDI")
+    image_mdi.attrib["key"] = "INTERLEAVE"
+    image_mdi.text = "PIXEL"
+    for idx, band in enumerate(["RED", "GREEN", "BLUE"]):
```

Comment:
test for number of bands to be 3 or 4 (dealing with the 4th band can be leaved for later)